### DISCUSSION TITLE
How can I make an image transparent?

### DISCUSSION THREAD

**Discussion by ThatRendle** [SixLabors/ImageSharp#1272] [Q&A] [answered]
I've got an image created from a document with a white background, in PNG format. I guess there must be some way to replace the white with transparent, but I don't know where to start.

**Comment by JimBobSquarePants** [comment, marked as answer]
There's a few ways:

1. If your image is not antialiased you could loop through the pixels and replace each instance of `Color.White` with `Color.Transparent`.
2. Do the above but with some sort of threshold to capture variance.
3. Use the [`ImageSharp.Drawing`](https://www.nuget.org/packages/SixLabors.ImageSharp.Drawing/1.0.0-beta0010) package and  [`RecolorBrush`](https://docs.sixlabors.com/api/ImageSharp.Drawing/SixLabors.ImageSharp.Drawing.Processing.RecolorBrush.html) which takes source and target colors. You then pass that to the [`Fill`](https://docs.sixlabors.com/api/ImageSharp.Drawing/SixLabors.ImageSharp.Drawing.Processing.FillExtensions.html#SixLabors_ImageSharp_Drawing_Processing_FillExtensions_Fill_IImageProcessingContext_SixLabors_ImageSharp_Drawing_Processing_IBrush_) method.

I think 3 is probably the easiest way to go, bear in mind though that `ImageSharp.Drawing` is a beta and subject to change. 

``` c#
using var image = Image.Load(...);

float threshold = 0..1F;
Color sourceColor = Color.White;
Color targetColor = Color.Transparent;
var brush = new RecolorBrush(sourceColor, targetColor, threshold);

image.Mutate(x => x.Fill(brush));

image.Save(...);

**Reply by ThatRendle** [reply]
Thanks Jim. I couldn't get the RecolorBrush approach to work, but figured out looping through the pixels and that works fine. The resulting image is going to be overlaid on another white background image so the anti-aliasing shouldn't be a problem.

**Reply by Tairitsua** [reply]
Thanks! but RecolorBrush couldn't work when set its target color as Transparent. I tried to make target color like Red or other color and changed threshold it worked fine. Why?

**Reply by JimBobSquarePants** [reply]
You need to set `PixelAlphaCompositionMode.Clear` for the `AlphaCompositionMode` in your `GraphicsOptions`

See https://github.com/SixLabors/ImageSharp.Drawing/discussions/147

**Reply by Tairitsua** [reply]
Thanks you! But if I set `PixelAlphaCompositionMode.Clear` for the `AlphaCompositionMode `, my out put will become totally transparent no matter how I change the threshold.
Then I found SixLabors/ImageSharp.Drawing#26, and I set it as `PixelAlphaCompositionMode.Src` and it works for me. (btw, I forgot to set the PngColorType to RgbWithAlpha in the png encoder at the begining)

**Reply by JimBobSquarePants** [reply]
I'd like to see If I can make that a default for `RecolorBrush`. It's tricky though.

**Reply by mamdos** [reply]
what if my actual image has white pixels and i don't want to lose it , i just want to replace the background white color to transparent not the actual content itself !
